crates/apollo-smith/src/enum_.rs

```diff
@@ -40,6 +48,58 @@ impl From<EnumTypeDef> for EnumDefinition {
     }
 }
 
+#[cfg(feature = "parser-impl")]
+impl From<apollo_parser::ast::EnumTypeDefinition> for EnumTypeDef {
+    fn from(enum_def: apollo_parser::ast::EnumTypeDefinition) -> Self {
+        Self {
+            description: enum_def
+                .description()
+                .map(|d| Description::from(d.to_string())),
+            name: enum_def.name().unwrap().into(),
+            directives: enum_def
+                .directives()
+                .map(|d| {
+                    d.directives()
+                        .map(|d| (d.name().unwrap().into(), Directive::from(d)))
+                        .collect()
+                })
+                .unwrap_or_default(),
+            enum_values_def: enum_def
+                .enum_values_definition()
+                .expect("must have enum values definition")
+                .enum_value_definitions()
+                .map(EnumValueDefinition::from)
+                .collect(),
+            extend: false,
+        }
+    }
+}
+
+#[cfg(feature = "parser-impl")]
+impl From<apollo_parser::ast::EnumTypeExtension> for EnumTypeDef {
+    fn from(enum_def: apollo_parser::ast::EnumTypeExtension) -> Self {
+        Self {
+            description: None,
+            name: enum_def.name().unwrap().into(),
```

Comment:
At this line of the codebase we don't really care about the name because we just assume the document parser by the apollo-parser is correct. BUT, it's useful to know that in different places, I will add some checks ! Thank you !

Reply:
Applied for all TypeExtensions

Reply:
`apollo-parser` does not do semantic validation, so it does not check whether an enum definition name already exists when bumping enum extension name. does that make sense?

or am i misunderstanding the `parser_impl` feature? you meant for it to be semantically correct as well?